examples/priviledges/main.tf

```diff
@@ -0,0 +1,192 @@
+# NOTE that this is an RFC for how things *could* work.
+
+# This is the schema for granting priviledges
+# https://docs.snowflake.net/manuals/sql-reference/sql/grant-privilege.html
+# 
+# GRANT {  { globalPrivileges | ALL [ PRIVILEGES ] } ON ACCOUNT
+#        | { accountObjectPrivileges | ALL [ PRIVILEGES ] } ON { RESOURCE MONITOR | WAREHOUSE | DATABASE } <object_name>
+#        | { schemaPrivileges | ALL [ PRIVILEGES ] } ON { SCHEMA <schema_name> | ALL SCHEMAS IN DATABASE <db_name> }
+#        | { schemaObjectPrivileges | ALL [ PRIVILEGES ] } ON { <object_type> <object_name> | ALL <object_type>S IN SCHEMA <schema_name> }  }
+#        | { schemaObjectPrivileges | ALL [ PRIVILEGES ] } ON FUTURE <object_type>S IN SCHEMA <schema_name> }  }
+#   TO [ ROLE ] <role_name> [ WITH GRANT OPTION ]
+
+# globalPrivileges ::=
+#   { { CREATE { ROLE | USER | WAREHOUSE | DATABASE } } | MANAGE GRANTS | MONITOR USAGE } [ , ... ] }
+
+# accountObjectPrivileges ::=
+# -- For RESOURCE MONITOR
+#   { MODIFY | MONITOR } [ , ... ]
+# -- For WAREHOUSE
+#   { MODIFY | MONITOR | USAGE | OPERATE } [ , ... ]
+# -- For DATABASE
+#   { MODIFY | MONITOR | USAGE | CREATE SCHEMA | IMPORTED PRIVILEGES } [ , ... ]
+
+# schemaPrivileges ::=
+#   { MODIFY | MONITOR | USAGE | CREATE { TABLE | VIEW | FILE FORMAT | STAGE | PIPE | SEQUENCE | FUNCTION } } [ , ... ]
+
+# schemaObjectPrivileges ::=
+# -- For TABLE
+#   { SELECT | INSERT | UPDATE | DELETE | TRUNCATE | REFERENCES } [ , ... ]
+# -- For VIEW
+#     SELECT
+# -- For internal STAGE
+#     READ [ , WRITE ]
+# -- For external STAGE
+#     USAGE
+# -- For FILE FORMAT, UDF, or SEQUENCE
+#     USAGE
+
+# Given that schema, here are some terraform resources we could support. We should map this out semi-fully, even though
+# we will build incrementally.
+
+# set global privs
+resource "snowflake_priviledge_grants" "a" {
+  name      = "foo1"
```

Comment:
does this name do anything other than provide an ID to track tf state?

Reply:
Just state, because there is no natural ID otherwise.